### DISCUSSION TITLE
Can you explain what you mean by this...

### DISCUSSION THREAD

**Discussion by bitsByB** [mohankumarpaluru/oracle-freetier-instance-creation#48] [General]
hi @mohankumarpaluru,
from the pre-requisites you say: 
> VM.Standard.E2.1.Micro Instance: The script is designed for a Ubuntu environment, and you need an existing subnet ID for ARM instance creation. Create an always-free VM.Standard.E2.1.Micro instance with Ubuntu 22.04. This instance can be deleted after the ARM instance creation. (Not required if an existing OCI_SUBNET_ID is defined in oci.env file)

Let's break it down,

1. how do I do this?
> you need an existing subnet ID for ARM instance creation

2. Does it mean I'll have 2 instances running once the ARM one is created? Could you explain how it works 
> Create an always-free VM.Standard.E2.1.Micro instance with Ubuntu 22.04

3. Could you explain this in more detail please?
> This instance can be deleted after the ARM instance creation. (Not required if an existing OCI_SUBNET_ID is defined in oci.env file)

thanks a lot @mohankumarpaluru, I'm still trying to have your script running but there's still some stuff I don't understand.
:pray:

**Comment by LordSithek** [comment]
As of my current knowledge:

1. The instruction for finding subnet ID are in the [README page](https://github.com/mohankumarpaluru/oracle-freetier-instance-creation?tab=readme-ov-file#environment-variables) of this script. Look under `OCI_SUBNET_ID`. Basically, you go to Oracle console and find the place given in the instruction.
2. I believe yes, you should have two instances and you can delete the Micro one after you get Ampere. But if I understand Oracle terms of service corretly, the total storage of all your instances is limited to 200GB, so if you create the Micro instance with min. of 50GB storage, you have max. 150GB left for Ampere instance. That's why I didn't want to set up the script on the Micro instance, rather I did it on the separate Virtual Machine created on Google Cloud Platform.
3. If I understand correctly, when you run the script on OCI Micro instance, your subnet ID is defined by default. But if you run the script elsewhere, like me, you must manually fill `OCI_SUBNET_ID` variable in order to your new Ampere instance could be created.

Perhaps @mohankumarpaluru corrects me if I'm wrong

**Reply by bitsByB** [reply]
Thanks for taking the time @LordSithek, I appeciate it :+1:

**Comment by LordSithek** [comment]
By the way, the more I read the web, the more I'm convinced that in the current situation Pay-As-You-Go is the way to go :) There is a nice explanation [here](https://www.youtube.com/watch?v=Xto9zROBwqY) about how to set budget alerts on PAYG to prevent unexpected costs
